apis/applications/v1alpha1/types.go

```diff
@@ -55,6 +55,10 @@ type ApplicationParameters struct {
 
 	// SourceHydrator provides a way to push hydrated manifests back to git before syncing them to the cluster.
 	SourceHydrator *SourceHydrator `json:"sourceHydrator,omitempty" protobuf:"bytes,14,opt,name=sourceHydrator"`
+
+	// AppNamespace is the namespace of the application in the ArgoCD server
+	// +optional
+	AppNamespace string `json:"appNamespace"`
```

Comment:
Optional fields should be pointers and have the `omitempty` json tag. The `+optional` is not necessary then because it is implied.

```suggestion
	AppNamespace *string `json:"appNamespace,omitempty"`
```